src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtService.java

```diff
@@ -144,15 +152,38 @@ private boolean checkPilotCourt(PilotCourt pilotCourt, String courtName, Pilot p
         return pilotCourts;
     }
 
-    private Set<Pilot> getPilots(CSVRecord next) {
-        Set<Pilot> pilots = new HashSet<>();
-        if (Boolean.parseBoolean(next.get(PilotCourtCSVHeader.LA_PILOT.ordinal()))) {
-            pilots.add(Pilot.LA);
-        }
-        if (Boolean.parseBoolean(next.get(PilotCourtCSVHeader.JDDO_PILOT.ordinal()))) {
-            pilots.add(Pilot.JDDO);
+    private Map<Pilot, LocalDateTime> getPilots(CSVRecord csvRecord) {
+        return Arrays.stream(PilotCourtCSVHeader.values())
+        .filter(pilotCourtCSVHeader -> pilotCourtCSVHeader.getPilot().isPresent())
+        .map(pilotCourtCSVHeader -> {
+            String pilotConfig = csvRecord.get(pilotCourtCSVHeader.ordinal());
+
+            if (!isValidPilotDate(pilotConfig)) {
+                return null;
+            }
+
+            return new Object[]{
+                pilotCourtCSVHeader.getPilot().get(),
+                LocalDate.parse(pilotConfig, ISO_LOCAL_DATE).atTime(PILOT_COURT_GO_LIVE_TIME, 0)
+            };
+
+        })
+        .filter(Objects::nonNull)
+        .collect(Collectors.toMap(e -> (Pilot)e[0], e -> (LocalDateTime) e[1]));
+    }
+
+    // Valid configuration for a pilot trigger date is either a date in IOS format (yyyy-MM-dd) or false
+    private boolean isValidPilotDate(String pilotConfig) {
+        try {
+            LocalDate.parse(pilotConfig, ISO_LOCAL_DATE);
+        } catch (DateTimeParseException e) {
+            if (!pilotConfig.equals("false")) {
+                throw new IllegalArgumentException(String.format("Expecting configuration to either be a valid date or "
+                    + "'false' but got %s", pilotConfig));
```

Comment:
Could we possibly detect malformatting earlier in the PostConstruct process? Regex perhaps. Then this part of the code could be cleaner.

Reply:
Regex is not clean!

Reply:
Personally I think reading and validating in a single pass is better.